net/upnp/src/www/status_upnp.php

```diff
@@ -62,41 +62,42 @@
 <?php
           else: ?>
           <div class="table-responsive">
-            <table class="table table-striped">
+            <table class="table table-striped table-condensed table-hover">
               <thead>
                 <tr>
-                  <td><?=gettext("Port");?></td>
-                  <td><?=gettext("Protocol");?></td>
-                  <td><?=gettext("Internal IP");?></td>
-                  <td><?=gettext("Int. Port");?></td>
-                  <td><?=gettext("Description");?></td>
+                  <td><?=gettext("Interface")?></td>
+                  <td><?=gettext("Ext. Port")?></td>
+                  <td><?=gettext("Internal IP")?></td>
+                  <td><?=gettext("Int. Port")?></td>
+                  <td><?=gettext("Protocol")?></td>
+                  <td><?=gettext("Source IP")?></td>
+                  <td><?=gettext("Source Port")?></td>
+                  <td><?=gettext("Description")?></td>
                 </tr>
               </thead>
               <tbody>
 <?php
               foreach ($rdr_entries as $rdr_entry):
-                  if (!preg_match("/on (.*) inet proto (.*) from (.*) to (.*) port = (.*) keep state label \"(.*)\" rtable [0-9] -> (.*) port (.*)/", $rdr_entry, $matches)) {
+                  if (!preg_match("/on (?P<iface>.*) inet proto (?P<proto>.*) from (?P<srcaddr>.*) (port (?P<srcport>.*) )?to (?P<extaddr>.*) port = (?P<extport>.*) keep state (label \"(?P<descr>.*)\" )?rtable [0-9] -> (?P<intaddr>.*) port (?P<intport>.*)/", $rdr_entry, $matches)) {
                       continue;
                   }
-                  $rdr_proto = $matches[2];
-                  $rdr_port = $matches[5];
-                  $rdr_label =$matches[6];
-                  $rdr_ip = $matches[7];
-                  $rdr_iport = $matches[8];
               ?>
                 <tr>
-                  <td><?=$rdr_port;?></td>
-                  <td><?=$rdr_proto;?></td>
-                  <td><?=$rdr_ip;?></td>
-                  <td><?=$rdr_iport;?></td>
-                  <td><?=$rdr_label;?></td>
+                  <td><?= htmlspecialchars(convert_friendly_interface_to_friendly_descr(convert_real_interface_to_friendly_interface_name($matches['iface']))) ?></td>
```

Comment:
It's not very elegant but the best thing to do here for now.  please use `html_safe` instead of `htmlspecialchars` everywhere